addon/jsonapi-attribute-transformer.js

```diff
@@ -47,21 +46,23 @@ var JSONAPIAttributeTransformer = function (/*store*/) {
    @param modelName
    @param fixture
    */
-  var transformSingle = function (fixture) {
-    transformAttributes(fixture);
-    findRelationships(fixture);
+  var transformSingle = function (modelName, fixture) {
+    transformAttributes(modelName, fixture);
+    findRelationships(modelName, fixture);
   };
 
-  var transformAttributes = function (fixture) {
+  var transformAttributes = function (modelName, fixture) {
     if (fixture.attributes) {
-      transormObjectKeys(fixture.attributes);
+      transformObjectKeys(modelName, fixture.attributes);
     }
   };
 
-  var transormObjectKeys = function(object) {
+  var transformObjectKeys = function(modelName, object) {
+    var serializer = store.serializerFor(modelName);
+    var transformFunction = serializer.keyForAttribute || Ember.String.dasherize;
```

Comment:
Simply put, we now use the serializer for a specific model name and it's `keyForAttribute` function. I still fallback to `dasherize` by default, but that should never happen, since there's always at least the default serializer the app uses.